instrumentation/spring/spring-integration-4.1/javaagent/src/test/groovy/SpringIntegrationAndRabbitTest.groovy

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+import static io.opentelemetry.api.trace.SpanKind.SERVER
+import static io.opentelemetry.instrumentation.test.server.ServerTraceUtils.runUnderServerTrace
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+
+class SpringIntegrationAndRabbitTest extends AgentInstrumentationSpecification implements WithRabbitProducerConsumerTrait {
+  def setupSpec() {
+    startRabbit()
+  }
+
+  def cleanupSpec() {
+    stopRabbit()
+  }
+
+  def "should cooperate with existing RabbitMQ instrumentation"() {
+    when:
+    // simulate the workflow being triggered by HTTP request
+    runUnderServerTrace("HTTP GET") {
+      producerContext.getBean("producer", Runnable).run()
+    }
+
+    then:
+    assertTraces(2) {
+      trace(0, 7) {
+        span(0) {
+          name "HTTP GET"
+          kind SERVER
+          attributes {}
+        }
+        span(1) {
+          name "producer"
+          childOf span(0)
+          attributes {}
+        }
+        span(2) {
+          name "exchange.declare"
+          childOf span(1)
+          kind CLIENT
+          attributes {
+            "${SemanticAttributes.NET_PEER_NAME.key}" { it == null || it == "localhost" }
+            "${SemanticAttributes.NET_PEER_IP.key}" "127.0.0.1"
+            "${SemanticAttributes.NET_PEER_PORT.key}" { it == null || it instanceof Long }
+            "${SemanticAttributes.MESSAGING_SYSTEM.key}" "rabbitmq"
+            "${SemanticAttributes.MESSAGING_DESTINATION_KIND.key}" "queue"
+          }
+        }
+        span(3) {
+          name "testTopic -> testTopic send"
```

Comment:
I think messaging semantic conventions suggest this should be just `testTopic send`

Reply:
That span comes from rabbit instrumentation, do you mind if I create an issue for this?

Reply:
It seems that we should review rabbitmq integration's usage of semantic convention anyway. There are some other strange things here

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/3337